python/actions/clone_review_with_latest/hook/clone_review.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+# :coding: utf-8
+# :copyright: Copyright (c) 2018 ftrack
+
+import logging
+import ftrack_api
+import datetime
+from ftrack_action_handler.action import BaseAction
+
+
+class CloneReview(BaseAction):
+    '''Clone review class.'''
+
+    label = 'Clone review with Latest versions'
+    identifier = 'com.ftrack.recipes.clone_review'
+    description = 'Clone review and update with latest versions available.'
+    
+    def validate_selection(self, entities):
+        '''Return True if the selection is valid.
+
+        Utility method to check *entities* validity.
+
+        '''
+        if not entities:
+            return False
+
+        if len(entities) > 1:
+            # works on one at time only.
+            return False
+        
+        entity_type, entity_id = entities[0]
+        if entity_type == 'ReviewSession':
+            return True
+
+        return False
+    
+    def discover(self, session, entities, event):
+        '''Return True if the action can be discovered.
+
+        Check if the current selection can discover this action.
+
+        '''
+        return self.validate_selection(entities)
+
+    def launch(self, session, entities, event):
+        '''Return result of running action.'''
+
+        for entity_type, entity_id in entities:
+            review_session = session.get(entity_type, entity_id)
+            new_review_objects = []
+            objects = review_session['review_session_objects']
+            for object in objects:
+                latest_version  = session.query(f'AssetVersion where asset_id is {object["asset_version"]["asset_id"]} and is_latest_version is True').one()
+                new_session_object = session.create('ReviewSessionObject', {'asset_version': latest_version})
+                new_review_objects.append(new_session_object)
+
+            review_name = datetime.datetime.now().strftime("%d/%m/%Y")
+            new_list = session.create(
+                'ReviewSession', 
+                   {
+                       'name': review_name,
```

Comment:
I would probably have given it something more descriptive as the name but this is OK :)

Reply:
keeping it simple for now, is just an example ;)